### PR TITLE
Merge release 0.42.9

### PR DESCRIPTION
**Pull request by iproudhon**
Merge release 0.42.9

## Description
Merging cosmos-sdk release 0.42.9 to stay up-to-date. This doesn't include any major changes, mostly fixes.

Other than fixes, this PR includes following changes.

commit 9db413bc80bdcc3768f42728e22a92433c74b9db
    feat: Query txs by signature and by address+seq (backport #9750) (#9783)

commit 6c49c4a55ca2a28fde795fc4774732c302c4eb30
    feat: Error on blank chain-id in multisign command (backport #9593) (#9605)

commit 919ae01b3b61daa3a93de7a02e34f947da7007fd
    feat: return trace value from baseapp (backport #9578) (#9580)

commit d773c88359ea10ad79070951aaf67940114ac647
    feat: add `RefundGas` function to `GasMeter` (backport #9403) (#9444)

commit 850da8a8b66dd29e80a073fb62e749f1fe7f5cd7
    feat: add header hash to `Context` (backport #9390) (#9395)

commit 3550f15ac4d206115294886d37044339d3417e12
    CLI: `query ibc-transfer escrow-address` (#9383)

commit 5531aaf9346f1b93fcc1d2b4951c6743e023b757
    Add env variable to cmd flags (backport #9040) (#9318)

commit 80a99b3265ee01f8402154bbea69a79f752e1d6b
    backport: make NewDecFromStr returns error for too large decimal (#9175)

commit aba621cbcc48a94a1bf977b34fecb76a568a95c4
    crypto/types: check for overflow and unreasonably large element count (#9173)

commit 1e03826d72d1ee1f6384f71d2d8d4c21cde38a34
    [Backport] Add ledger/multisig detection in SignTx functions (#9026) (#9041)

commit 177672fd147c3924dbe6df8dcb253524717c30b0
    add orderBy parameter to TxsByEvents (bp #8815) (#8883)

commit e23c0ce3e647630f27bf3ec944dab77f24f68d35
    add --output-document to multisign-batch (#8873) (#8877)

commit 2ffe5303734a451a38fd96c280ed2d058bad51fe
    Security/ghsa mvm6 gfm2 89xj (#8852)

commit c9f36a850a94600c1b8ab12320d0acfedc962966
    * fix multisig account pubkeys migration (#8794)
    * Supply by denom Migrations (#8780)
    * Add back supply proto
    * Add migration for supply
    * reduce gas costs by 10x for transient store operations (#8790)
    * fix TestTransientGasConfig for ReadCostFlat
    * x/gov: fix NormalizeProposalType() return values (#8808)
    * store/cachekv: use typed types/kv.List instead of container/list.List (#8811)

commit 1391ea624f4acf706b45a119f15554fd86fc7c87
    store/cachekv: use typed types/kv.List instead of container/list.List (#8811) (#8817)

commit 339484a728b0917db82b9c2552160814097ed9f1
    add trust to macOS Keychain for calling apps by default (#8826) (#8835)

commit be34d525ba42c607ea866ffe95dc8910888ab03b
    Add client config subcommand to CLI (backport #8953) (#9255)

## Checklist:
- [ ] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: store/iavl/prefetch.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		prefetchLocks <- true
+		go func(f func()) {
+			f()
+			<-prefetchLocks
+		}(f)
+	}
+}
+
+// Implements type.KVStore.
```
Comment (zemyblue) [inline store/iavl/prefetch.go:65]: lint warning
so `// Prefetch implements type.KVStore.`

**Review by zemyblue — COMMENTED** [review]
Thank you for your checking out the code, even though the code is very huge.
I add some comments where I don't understand.

**Comment by zemyblue** [comment]
And can we ignore codecoverage errors?

File: store/cachemulti/store.go [hunk 2]
```diff
@@ -110,8 +111,26 @@ func (cms Store) GetStoreType() types.StoreType {
...
+				}
+				wg.Done()
+			}()
+
+			s.Write()
+		}(store)
+	}
+	wg.Wait()
+	if panicMsg != nil {
+		panic(panicMsg)
```
[the remark this answers is not in the document]
Comment (iproudhon) [reply store/cachemulti/store.go:133]: It's most likely captured by caller.

**Review by zemyblue — APPROVED** [review]
LGTM

**Review by dudong2 — APPROVED** [review]

**Pull request merged**